### DISCUSSION TITLE
Adding AI to RapidPro

### DISCUSSION THREAD

**Discussion by micuentadecasa** [nyaruka/rapidpro#5258] [Ideas]
Hi, I´m evaluatiing how to add AI to RapidPro, I have seen some code from two years ago for using the "hotbot", but it seems nothing has changed since then, so I would like to know what is the recommended way to add some NLP, AI functions to RapidPro?

Regards.

**Comment by ericnewcomer** [comment]
Yes, this has been a common topic of late!

You could certainly pull in resources (such as OpenAI) using a webhook action. I've certainly done this. That said, we are exploring a deeper integration that allows LLMs to be used in more places throughout the platform. For example, summarizing responses and a node in a flow or using LLMs for classification/routing. We've so far identified that to enable these deeper integrations suggests a re-architecture of our storage model to support some new access patterns that would be required -- particularly at scale.

**Comment by rowanseymour** [comment]
10.2 now has support for adding different LLM services and then using those to translate flows or in new Call LLM actions within flows.
